public/src/components/switchboard.tsx

```diff
@@ -1,456 +1,211 @@
-      enableContributionsCampaign: SwitchState.Off,
-      forceContributionsCampaign: SwitchState.Off,
-      enableQuantumMetric: SwitchState.Off,
-      experiments: {},
-    };
-    this.previousStateFromServer = null;
-  }
-
-  UNSAFE_componentWillMount(): void {
-    this.fetchStateFromServer();
-  }
-
-  fetchStateFromServer(): void {
-    fetchSupportFrontendSettings(SupportFrontendSettingsType.switches).then(serverData => {
-      this.previousStateFromServer = serverData;
-      this.setState({
-        ...serverData.value,
-      });
-    });
-  }
-
-  updateOneOffPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        oneOffPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateRecurringPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        recurringPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateFeatureSwitch(switchName: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        experiments: {
-          [switchName]: {
-            state: { $set: switchState },
-          },
-        },
-      }),
+    updatedState[groupId].switches[mySwitchId].state = event.target.checked
+      ? SwitchState.On
+      : SwitchState.Off;
+
+    setData(updatedState);
+    setNeedToSaveDataWarning(true);
+  };
+
+  const createSwitchesFromGroupData = (
+    mySwitch: [string, Switch],
+    groupId: string,
+  ): JSX.Element => {
+    const [switchId, switchData] = mySwitch;
+
+    return (
+      <FormControlLabel
+        label={switchData.description}
+        checked={switchData.state === 'On' ? true : false}
+        control={<SwitchUI />}
+        onChange={(event): void => updateSwitchSetting(groupId, mySwitch, event)}
+        value={switchId}
+        key={switchId}
+      />
     );
-  }
-
-  saveSwitches = (): void => {
-    const newState = update(this.previousStateFromServer, {
-      value: { $set: this.state },
-    });
-
-    saveSupportFrontendSettings(SupportFrontendSettingsType.switches, newState)
-      .then(resp => {
-        if (!resp.ok) {
-          resp.text().then(msg => alert(msg));
-        }
-        this.fetchStateFromServer();
-      })
-      .catch(resp => {
-        alert(`Error while saving: ${resp}`);
-        this.fetchStateFromServer();
-      });
   };
 
-  render(): React.ReactNode {
-    const { classes } = this.props;
+  const createGroupsFromData = (group: [string, SwitchGroup]): JSX.Element => {
+    const [groupId, groupData] = group;
 
     return (
+      <FormControl className={classes.formControl} key={groupId}>
+        <FormLabel>{groupData.description}</FormLabel>
+        {Object.entries(groupData.switches).map(mySwitch =>
```

Comment:
I think it's best to destructure tuples asap, it saves having to do `switchData[0]` etc elsewhere. E.g.
```
Object.entries(groupData.switches).map(([switchName, switchData]) =>
  createSwitchesFromGroupData(switchName, switchData, groupId),
)
```